services/common/src/main/java/org/finra/gatekeeper/GatekeeperCommonConfig.java

```diff
@@ -162,6 +167,11 @@ public LdapTemplate ldapTemplate(ContextSource contextSource) {
     @Bean
     public GatekeeperAuthorizationService gatekeeperLDAPAuthorizationService(LdapTemplate ldapTemplate,
                                                                              Supplier<IGatekeeperUserProfile> gatekeeperUserProfileSupplier){
+        //User Headers for Auth
+        if(gatekeeperAuthProperties.isUseHeadersForMembership()){
```

Comment:
I Think we should re-work this to be a switch statement based off a value, for example we could call the new property "method" and support the following values: "header", "activeDirectory", "openLdap"